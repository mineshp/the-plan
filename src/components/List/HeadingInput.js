import React from 'react';
import PropTypes from 'prop-types';
import { Button, Divider, Form, Grid, Input } from 'semantic-ui-react';

const HeadingInput = ({ headings, handleAddColumn, handleRemoveColumn, handleHeaderInputChange }) => {
    const headerRows = [];
    headings.map((heading) => {
        const headingID = heading.id; // eslint-disable-line no-underscore-dangle
        return (
            headerRows.push(
                <Grid.Row key={headingID}>
                    <Grid.Column>
                        <Form.Field>
                            <Input
                                placeholder="Header Name..."
                                name={heading.name}
                                id={headingID}
                                key={headingID}
                                defaultValue={heading.name}
                                onChange={handleHeaderInputChange}
                            />
                        </Form.Field>
                    </Grid.Column>
                    <Grid.Column>
                        <Button
                            icon="trash"
                            content="Remove"
                            labelPosition="left"
                            id={headingID}
                            color="pink"
                            type="button"
                            onClick={handleRemoveColumn}
                            floated="left"
                        />
                    </Grid.Column>
                </Grid.Row>
            )
        );
    });

    return (
        <div>
            <Grid columns={2} divided>
                <Divider horizontal>ASSIGN LIST HEADINGS</Divider>
                {headerRows}
                <Divider />
                <Button fluid type="button" color="blue" onClick={handleAddColumn}>Add Column</Button>
            </Grid>
        </div>
    );
};

HeadingInput.propTypes = {
    headings: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string,
            name: PropTypes.string
        }).isRequired
    ).isRequired,
    handleAddColumn: PropTypes.func.isRequired,
    handleRemoveColumn: PropTypes.func.isRequired,
    handleHeaderInputChange: PropTypes.func.isRequired
};

export default HeadingInput;
