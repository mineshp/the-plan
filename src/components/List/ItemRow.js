import React from 'react';
import PropTypes from 'prop-types';
import { Button, Table } from 'semantic-ui-react';


const ItemRow = ({ data, listID }) => (
    <Table.Row>
        <Table.Cell>{data.name}</Table.Cell>
        <Table.Cell>{data.createdDate}</Table.Cell>
        <Table.Cell>{data.updatedDate}</Table.Cell>
        <Table.Cell>{data.notes}</Table.Cell>
        <Table.Cell>{data.owner}</Table.Cell>
        <Table.Cell><Button as='a' color='teal' href={`/list/${listID}/item/${data.id}/delete`}>Delete</Button></Table.Cell>
    </Table.Row>
);

ItemRow.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    createdDate: PropTypes.string.isRequired,
    updatedDate: PropTypes.string.isRequired
    }).isRequired
}

export default ItemRow;