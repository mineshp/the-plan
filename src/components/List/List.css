.ui.input.stdInput {
    width: 90% !important;
}

.ui.formSpacing {
    margin-top: 40px;
}