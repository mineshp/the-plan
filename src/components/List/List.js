import React from 'react';
import PropTypes from 'prop-types';
import { Container, Table } from 'semantic-ui-react';
import DisplayMessage from '../Shared/DisplayMessage';
import ListHeadings from './ListHeadings';
import ListItems from './ListItems';

const List = ({ list, errors }) => (
    <div className="List main">
        {
            (errors && errors.error) && <DisplayMessage status={errors} />
        }
        <Container>
            <Table celled striped>
                <ListHeadings headings={list.headings} />
                <ListItems items={list.items} headings={list.headings} />
            </Table>
        </Container>
    </div>
);

List.propTypes = {
    list: PropTypes.shape({}).isRequired,
    errors: PropTypes.shape({})
};

List.defaultProps = {
    list: {
        projects: [],
        headings: [],
        items: []
    },
    errors: {}
};

export default List;
