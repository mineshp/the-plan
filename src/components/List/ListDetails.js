import React from 'react';
import PropTypes from 'prop-types';
import Table from 'semantic-ui-react/dist/commonjs/collections/Table';
import { formatDate } from '../../helpers/validators/common';

const ListDetails = ({ list }) => {
    const owner = list.owner ? list.owner : 'unknown';
    return (
        <Table>
            <Table.Body>
                <Table.Row>
                    <Table.Cell className="details-header">Owner:</Table.Cell>
                    <Table.Cell>{owner}</Table.Cell>
                    <Table.Cell className="details-header">Created Date:</Table.Cell>
                    <Table.Cell>{formatDate(list.createdDate)}</Table.Cell>
                    <Table.Cell className="details-header">Last Updated:</Table.Cell>
                    <Table.Cell>{formatDate(list.updatedDate)}</Table.Cell>
                </Table.Row>
            </Table.Body>
        </Table>
    );
};

ListDetails.propTypes = {
    list: PropTypes.shape({}).isRequired
};

export default ListDetails;
