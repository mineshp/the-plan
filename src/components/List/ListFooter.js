import React from 'react';
import PropTypes from 'prop-types';
import { Table, Button, Icon } from 'semantic-ui-react';

const ListFooter = ({ numColumns, handleAddItem, handleSubmit }) => (
    <Table.Footer fullWidth>
        <Table.Row>
            <Table.HeaderCell colSpan={numColumns}>
                <Button floated="left" color="blue" icon labelPosition="left" size="small" onClick={handleAddItem}>
                    <Icon name="plus" /> Add Item
                </Button>
                <Button
                    floated="right"
                    icon
                    labelPosition="left"
                    color="green"
                    size="small"
                    onClick={handleSubmit}
                >
                    <Icon name="save" /> Save
                </Button>
            </Table.HeaderCell>
        </Table.Row>
    </Table.Footer>
);

ListFooter.propTypes = {
    numColumns: PropTypes.number.isRequired,
    handleAddItem: PropTypes.func.isRequired,
    handleSubmit: PropTypes.func.isRequired
};

export default ListFooter;
