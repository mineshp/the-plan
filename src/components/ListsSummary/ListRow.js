import React from 'react';
import PropTypes from 'prop-types';
import { Button, Table, Icon, Label } from 'semantic-ui-react';

const projectLabels = (projects) => projects.map((project) => (
    <Label
        as="a"
        key={project.id}
        color="red"
    >
        <Icon name="empty star" />
        {project.name}
    </Label>
));

const List = ({ data }) => {
    const listID = data._id; // eslint-disable-line no-underscore-dangle
    return (
        <Table.Row>
            <Table.Cell>{data.listName}</Table.Cell>
            <Table.Cell>{data.createdDate}</Table.Cell>
            <Table.Cell>{data.updatedDate}</Table.Cell>
            <Table.Cell>{projectLabels(data.projects)}</Table.Cell>
            <Table.Cell><Button as="a" href={`/list/view/${listID}`}>View</Button></Table.Cell>
        </Table.Row>
    );
};

List.propTypes = {
    data: PropTypes.shape({
        _id: PropTypes.string.isRequired,
        listName: PropTypes.string.isRequired
    }).isRequired
};

export default List;
