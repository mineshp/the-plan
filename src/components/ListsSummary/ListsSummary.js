import React from 'react';
import PropTypes from 'prop-types';
import { Container, Table } from 'semantic-ui-react';
import DisplayMessage from '../Shared/DisplayMessage';

const ListsSummary = ({
    errors,
    rows
}) => (
    <div className="List main">
        {
            (errors && errors.error) &&
                <DisplayMessage status={errors} />
        }
        <Container>
            <Table celled striped>
                <Table.Header>
                    <Table.Row>
                        <Table.HeaderCell>Name</Table.HeaderCell>
                        <Table.HeaderCell>Created</Table.HeaderCell>
                        <Table.HeaderCell>Last Updated</Table.HeaderCell>
                        <Table.HeaderCell>Projects</Table.HeaderCell>
                        <Table.HeaderCell>Actions</Table.HeaderCell>
                    </Table.Row>
                </Table.Header>

                <Table.Body>
                    {rows}
                </Table.Body>
            </Table>
        </Container>
    </div>
);

ListsSummary.propTypes = {
    errors: PropTypes.shape({}),
    rows: PropTypes.arrayOf(
        PropTypes.shape({})
    )
};

ListsSummary.defaultProps = {
    errors: null,
    rows: []
};

export default ListsSummary;
