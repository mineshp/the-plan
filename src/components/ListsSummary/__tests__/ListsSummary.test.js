import React from 'react';
import renderer from 'react-test-renderer';
import ListsSummary from '../ListsSummary';
import List from '../ListRow';
import { mockAllListsData } from '../../../helpers/test/testData/listData';

const mockAllLists = mockAllListsData();

const mockDeleteList = jest.fn();

const Rows = [];
mockAllLists.map((list) =>
    // eslint-disable-next-line no-underscore-dangle
    Rows.push(<List data={list} key={list._id} onDeleteHandler={mockDeleteList} />)
);

describe('List', () => {
    it('renders lists correctly', () => {
        const tree = renderer.create(<ListsSummary
            rows={Rows}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

