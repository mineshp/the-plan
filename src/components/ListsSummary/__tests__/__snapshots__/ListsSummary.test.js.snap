// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders lists correctly 1`] = `
<div
  className="List main"
>
  <div
    className="ui container"
  >
    <table
      className="ui celled striped table"
    >
      <thead
        className=""
      >
        <tr
          className=""
        >
          <th
            className=""
          >
            Name
          </th>
          <th
            className=""
          >
            Created
          </th>
          <th
            className=""
          >
            Last Updated
          </th>
          <th
            className=""
          >
            Projects
          </th>
          <th
            className=""
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        className=""
      >
        <tr
          className=""
        >
          <td
            className=""
          >
            HumanBody
          </td>
          <td
            className=""
          >
            2016-05-18T16:00:00Z
          </td>
          <td
            className=""
          >
            2016-05-18T16:00:00Z
          </td>
          <td
            className=""
          >
            <a
              className="ui red label"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="empty star icon"
              />
              biology
            </a>
            <a
              className="ui red label"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="empty star icon"
              />
              chemistry
            </a>
          </td>
          <td
            className=""
          >
            <a
              className="ui button"
              disabled={undefined}
              href="/list/view/001"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              View
            </a>
          </td>
        </tr>
        <tr
          className=""
        >
          <td
            className=""
          >
            Explorers
          </td>
          <td
            className=""
          >
            2016-10-8T13:16:00Z
          </td>
          <td
            className=""
          >
            2016-10-8T16:00:00Z
          </td>
          <td
            className=""
          >
            <a
              className="ui red label"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="empty star icon"
              />
              history
            </a>
          </td>
          <td
            className=""
          >
            <a
              className="ui button"
              disabled={undefined}
              href="/list/view/002"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              View
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`List renders successfully after error creating new project 1`] = `
<div
  className="List main"
>
  <div
    className="ui red message"
  >
    <p>
      oh no something bad happened
    </p>
  </div>
  <div
    className="ui container"
  >
    <table
      className="ui celled striped table"
    >
      <thead
        className=""
      >
        <tr
          className=""
        >
          <th
            className=""
          >
            Name
          </th>
          <th
            className=""
          >
            Created
          </th>
          <th
            className=""
          >
            Last Updated
          </th>
          <th
            className=""
          >
            Projects
          </th>
          <th
            className=""
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        className=""
      >
        <tr
          className=""
        >
          <td
            className=""
          >
            HumanBody
          </td>
          <td
            className=""
          >
            2016-05-18T16:00:00Z
          </td>
          <td
            className=""
          >
            2016-05-18T16:00:00Z
          </td>
          <td
            className=""
          >
            <a
              className="ui red label"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="empty star icon"
              />
              biology
            </a>
            <a
              className="ui red label"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="empty star icon"
              />
              chemistry
            </a>
          </td>
          <td
            className=""
          >
            <a
              className="ui button"
              disabled={undefined}
              href="/list/view/001"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              View
            </a>
          </td>
        </tr>
        <tr
          className=""
        >
          <td
            className=""
          >
            Explorers
          </td>
          <td
            className=""
          >
            2016-10-8T13:16:00Z
          </td>
          <td
            className=""
          >
            2016-10-8T16:00:00Z
          </td>
          <td
            className=""
          >
            <a
              className="ui red label"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="empty star icon"
              />
              history
            </a>
          </td>
          <td
            className=""
          >
            <a
              className="ui button"
              disabled={undefined}
              href="/list/view/002"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              View
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`List renders successfully after successful creation of project 1`] = `
<div
  className="List main"
>
  <div
    className="ui container"
  >
    <table
      className="ui celled striped table"
    >
      <thead
        className=""
      >
        <tr
          className=""
        >
          <th
            className=""
          >
            Name
          </th>
          <th
            className=""
          >
            Created
          </th>
          <th
            className=""
          >
            Last Updated
          </th>
          <th
            className=""
          >
            Projects
          </th>
          <th
            className=""
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        className=""
      >
        <tr
          className=""
        >
          <td
            className=""
          >
            HumanBody
          </td>
          <td
            className=""
          >
            2016-05-18T16:00:00Z
          </td>
          <td
            className=""
          >
            2016-05-18T16:00:00Z
          </td>
          <td
            className=""
          >
            <a
              className="ui red label"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="empty star icon"
              />
              biology
            </a>
            <a
              className="ui red label"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="empty star icon"
              />
              chemistry
            </a>
          </td>
          <td
            className=""
          >
            <a
              className="ui button"
              disabled={undefined}
              href="/list/view/001"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              View
            </a>
          </td>
        </tr>
        <tr
          className=""
        >
          <td
            className=""
          >
            Explorers
          </td>
          <td
            className=""
          >
            2016-10-8T13:16:00Z
          </td>
          <td
            className=""
          >
            2016-10-8T16:00:00Z
          </td>
          <td
            className=""
          >
            <a
              className="ui red label"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="empty star icon"
              />
              history
            </a>
          </td>
          <td
            className=""
          >
            <a
              className="ui button"
              disabled={undefined}
              href="/list/view/002"
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              View
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
